document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -482,6 +520,17 @@ private String prepareUpsertDocument(Key key, Document document) throws IOExcept
     return MAPPER.writeValueAsString(jsonNode);
   }
 
+  private String prepareCast(String field, Object value) {
+    String fmt = "CAST (%s AS %s)";
+    if (value instanceof Number) {
+      return String.format(fmt, field, "NUMERIC");
+    } else if (value instanceof Boolean) {
+      return String.format(fmt, field, "BOOLEAN");
+    } else /* default is string */ {
```

Comment:
What if the value is a List (IN clause case)

Reply:
I couldnt find a test case where the IN clause values are numbers. May be worth checking that case once.

Reply:
Good point. 

There are two ways to handle this case.
- Option1: For the `IN` operator, it applies equality operator and will not affect sorting order. So, we can cast the values to the string while addObjectParam. It will be the generic solution and can handle mixed types in List
- Option2: As another alternative, we can consider the first element of the List, and consider its type as cast type. It will also be a fair assumption that `IN` or any other operator will have the same type values. So, as consistency, we can have CAST ( <field name> as <type> ) for all non string operator.

I am going ahead with option (2) and will update the PR with change and test case.

Reply:
Made the changes, and updated the PR. Added this test case for `testInQueryWithNumberField` this specific scenario.